tests/test_core_network.py

```diff
@@ -1938,6 +1938,26 @@ def test_adding_multiple_links_with_id_and_multi_idx_clashes(assert_semantically
     )
 
 
+def test_adding_many_links_with_id_and_multi_idx_clashes(assert_semantically_equal, mocker):
```

Comment:
The name of the test doesn't help me understand the intent of the test.

How about: `test_avoids_destructive_index_clashes_when_adding_replacement_links`? Or... something better than that? :-)